### ISSUE TITLE
Manifest for playback of downloaded content

### ISSUE THREAD

**Issue by kgrevehagen** [google/ExoPlayer#8344]
When streaming and downloading dash(or hls for that matter) I sideload the manifest instead of passing it a url which contains the manifest. This of course works because I always provide the manifest by fetching it from our servers first.

However, I'm having problems understanding how this should work when trying to play the downloaded content, because then I will not get the manifest from our server(I might be offline etc).

Do I have to save the manifest so I can sideload it or is it enough to provide some sort of cache key to avoid this(and then the manifest will be created automatically)?

**Comment by marcbaechinger** [comment]
The [manifest is downloaded by the `SegmentDownloader` with a `CacheDataSource`](https://github.com/google/ExoPlayer/blob/8a0d52be4b5402af0987fc7d009bc49977c72b67/library/core/src/main/java/com/google/android/exoplayer2/offline/SegmentDownloader.java#L136) and is cached that way.

**Comment by kgrevehagen** [comment]
I'm sorry if I was a bit vague, but I'm sideloading an embedded manifest, not the manifest url. So it is not downloaded by the `SegmentDownloader` in my case. I added functionality to `SegmentDownloader` to achieve this. But anyways, I think I got my answer, and that is that I then have to save my manifest if I later want to play it offline, so that I don't have to fetch my embedded manifest once more.
Thanks!
